### PR TITLE
Fixing zeek SMB sessionization Issue

### PR DESCRIPTION
**Pull request by amanbansal2709**
Hi, we are requesting this change to highlight the SMB parsing issues when it only sees part of a session that results in the log file (smb_mapping.log file) being sent with no timestamp.

Detailed explanation: 
[Zeek SMB Sessionization Issue.pdf](https://github.com/zeek/zeek/files/5533843/Zeek.SMB.Sessionization.Issue.pdf)

Please let us know if you have any questions.

### CODE REVIEW

**Comment by JustinAzoff** [comment]
Does that need the calls to `copy` ?

**Comment by amanbansal2709** [comment]
Honestly, the 'copy' calls are probably excessive. I was hesitant to modify the data on the fly but from a performance perspective, I see how that is not optimal. The SMB parsing can do some strange things when we see packet loss or asymmetric routing and this was really just a band-aid that covered a really common failure case. We have spent some time trying to create pcap that can reproduce the issue and will include that if we manage to get it working.

File: scripts/base/protocols/smb/main.zeek [hunk 1]
```diff
@@ -225,7 +225,12 @@ function write_file_log(state: State)
 			else
 				add state$recent_files[file_ident];
 			}
-		
+
+		local log_f = copy(f);
+		if (!log_f?$ts){
+			log_f$ts = network_time()
```
Comment (sethhall) [inline scripts/base/protocols/smb/main.zeek:231]: The log_f variable is never used for anything here or in the other changes.  This set of changes wouldn't actually do anything.

**Review by sethhall — COMMENTED** [review]
I fully believe that there are bugs around missing packets in the SMB scripts, but unfortunately the approach taken in this change is only band aiding the actual problem and there seems to be some mistake in these changes anyway because the changes wouldn't actually do anything.

I think we may just have to go back to the beginning on this ticket because I don't think there is much we can do on this ticket without a pcap unfortunately.

**Comment by timwoj** [comment]
I'm taking this out of the 4.0.0 milestone and putting it in the 4.1.0 one since it seems incomplete.

File: scripts/base/protocols/smb/main.zeek (hunk 1 above)
Comment (amanbansal2709) [reply scripts/base/protocols/smb/main.zeek:231]: I realized now that I have missed changing the variable while writing into the log file. All three changes in the commit needed to be logged as shown below.

Log::write(FILES_LOG, log_f);

I am happy to do another PR for it and submit the pcap with it too.

**Comment by amanbansal2709** [comment]
I have committed another change to log the local variable into the writing function. I know it's just the band-aide for the issue and coping the calls is also not the most ideal solution here but that did patch the issue for us. Unfortunately, it's hard to provide the pcap here, having said that I also feel, if possible to find a way to declare the timestamps, if missing, "ts = network_time()" before logging it without copying the whole event.

**Comment by sethhall** [comment]
I addressed the problem differently in #1436

**Pull request closed without merging**